app/clients/sms/twilio.py

```diff
@@ -240,18 +251,27 @@ def translate_delivery_status(
             error_code = parsed_dict['ErrorCode'][0]
 
             if error_code in self.twilio_error_code_map:
-                notify_delivery_status = self.twilio_error_code_map[error_code]
+                notify_delivery_status: TwilioStatus = self.twilio_error_code_map[error_code]
             else:
-                notify_delivery_status = self.twilio_notify_status_map[twilio_delivery_status]
+                self.logger.warning(
+                    'Unaccounted for Twilio Error code: %s with message sid: %s',
+                    error_code,
+                    parsed_dict['MessageSid'][0],
+                )
+                notify_delivery_status: TwilioStatus = self.twilio_notify_status_map[twilio_delivery_status]
         else:
-            notify_delivery_status = self.twilio_notify_status_map[twilio_delivery_status]
-
-        translation = {
-            'payload': decoded_msg,
-            'reference': parsed_dict['MessageSid'][0],
-            'record_status': notify_delivery_status,
-        }
+            # Logic not being changed, just want to log this for now
+            if 'ErrorCode' in parsed_dict:
+                self.logger.warning('Error code: %s existed but status was not failed nor undelivered', error_code)
```

Comment:
We should also add `MessageSid` here.

Reply:
Yeah, that's a good call. I was just going to use it to add to the dictionary but it may be beneficial to see additional details.